Lecture 8 hw

Bepely opened the pull request:
Выполнил базовые требования по ДЗ. 
Не тратил время на контент и стили. Спешу нагнать учебный план. 

Починил prettier _(не прошло и года😓😓)_

Bepely committed:
Merge pull request #1 from Bepely/lecture-2-hw

Lecture 2 homework

Bepely committed:
Lecture 3 homework

• Added header and footer
• Added Layout
• Added Tabs functionality
• Reworked Restaurants display functionality

Bepely committed:
Lecture 3 homework 2

• Added tabs disable functionality
• Added constants Min and Max in counter
• Removed redundant fragment

Bepely committed:
Lecture 3 homework #3

• Updated button disabled functionality

Bepely committed:
Merge pull request #2 from Bepely/lecture-3-hw

Lecture 3 homework

Bepely committed:
Lecture 4 Homework

• Added useReducer hook
• Added ReviewForm component
• Added custom scroller

○ Reused counter (for no reason, kinda dumb)

Bepely committed:
Lecture 6 Homework

• Added edits for lecture 4
○Scroller mechanics fixed
○Review mechanics fixed
• Added CSS stylings for lecture 5 HW
• Added Dark/Light theme for lecture 6 HW
• Added user imitation for lecture 6 HW

Bepely committed:
Lecture 6 HW fixes

• States translated to context to avoid re-renders
• Created menuItem component
• Added minor CSS adjustments

Bepely committed:
Merge pull request #4 from Bepely/lecture-6-hw

Lecture 6 hw

Bepely committed:
Lecture 7 HW

• Added react-redux and RTK libraries
• Mock data Replaced with Normalized mock data
• Created a Store and slices for normalized data
• Replaced data handling from regular data to RTK

Bepely committed:
lecture-7-hw-2

• Reorganized file structure to have a components folder
• Improved RTK data usage. (from props drilling to selectors)

Bepely committed:
lecture-7-hw-3

• Added key props
• Reworked<li> element usage
• Reworked redux selectors usage

Bepely committed:
lecture-7-hw-4

• Updated key prop usage

Bepely committed:
lecture-8-hw

• Created Cart functionality
• Added React-dom-router and redone data flow according to it
• Prettified whole project

○ Didn't do much content and css, due to urgency

zubkov7 commented on src/components/cart/component.jsx:
```diff
@@ -0,0 +1,40 @@
+      <h2>Корзина</h2>
+      {userValue.value !== null ? (
+        <>
+          {items.length ? (
```
во вложенных условиях код не очень читается, лучше разнести по компонентам логику

zubkov7 commented on src/components/header/component.jsx:
```diff
@@ -0,0 +1,34 @@
+        <NavLink to="/">Доставка еды</NavLink>
+      </h1>
+
+      <div className={styles.loginWrapper}>
```
можно унести в отдельный компонент

zubkov7 commented on src/components/pageScroller/component.jsx:
```diff
@@ -0,0 +1,38 @@
+      style={{
+        width: `${scrollerWidth}%`,
+      }}
+    ></div>
```
можно одним тегом: `<div />`

zubkov7 commented on src/components/restarauntCartSection/component.jsx:
```diff
@@ -0,0 +1,31 @@
+  };
+
+  return (
+    <>
```
лишний фрагмент

zubkov7 commented on src/components/reviewForm/component.jsx:
```diff
@@ -0,0 +1,144 @@
+        {[...Array(5)].map((element, index) => {
+          const currentRating = index + 1;
+
+          return (
```
лучше унести в отдельный компонент

Bepely committed:
lecture-8-hw-2

• Removed unnecessary wrapper tags
• Divide huge components to sub-components (cart, header, review)
• Created "reducers" folder with formReducer file to separate reducer code from a component.
